README.md

```diff
@@ -1 +1,264 @@
-libsyscall_intercept - coming soon!
+# syscall_intercept
+
+Userspace syscall intercepting library.
+This repository is soon going to be deleted
+
+# How to build #
+
+Building libsyscall_intercept requires cmake.
+Example:
+```sh
+cmake path_to_syscall_intercept -DCMAKE_INSTALL_PREFIX=/usr -DCMAKE_BUILD_TYPE=Release -DCMAKE_C_COMPILER=clang
+make
+```
+alternatively:
+```sh
+ccmake path_to_syscall_intercept
+make
+```
+
+There is an install target. For now, all it does, is cp.
+```sh
+make install
+```
+
+Coming soon:
+```sh
+make test
+```
+
+# Synopsis #
+
+```c
+#include <libsyscall_intercept_hook_point.h>
+```
+```sh
+cc -lsyscall_intercept -fpic -shared source.c -o preloadlib.so
+
+LD_PRELOAD=preloadlib.so ./application
+```
+
+##### Description: #####
+
+The system call intercepting library provides a low-level interface
+for hooking Linux system calls in user space. This is achieved
+by hotpatching the machine code of the standard C library in the
+memory of a process. The user of this library can provide the
+functionality of almost any syscall in user space, using the very
+simple API specified in the libsyscall_intercept\_hook\_point.h header file:
+```c
+int (*intercept_hook_point)(long syscall_number,
+			long arg0, long arg1,
+			long arg2, long arg3,
+			long arg4, long arg5,
+			long *result);
+```
+
+The user of the library shall assign to the variable called
+intercept_hook_point a pointer to the address of a callback function.
+A non-zero return value returned by the callback function is used
+to signal to the intercepting library, that the specific system
+call was ignored by the user, and the original syscall should be
+executed. A zero return value signals that the user takes over the
+system call. In this case, the result of the system call
+(the value stored in the RAX register after the system call)
+can be set via the *result pointer. In order to use the library,
+the intercepting code is expected to be loaded using the
+LD_PRELOAD feature provided by the system loader.
+
+All syscalls issued by libc are intercepted. Syscalls made
```

Comment:
Just a question, nothing to change for now.  Is there a way to have the callbacks the user provided be a trigger as to which syscalls are intercepted, as opposed to having all syscalls intercepted? And, having the return value of the callback be the trigger as to whether or not we intercept the system call?